app/translators/tdr_manifest_to_rawls.py

```diff
@@ -160,9 +174,17 @@ def translate_parquet_row(self, row: pd.Series, column_names: List[str], is_azur
                     except json.JSONDecodeError:
                         logging.warning(f"Couldn't parse value {value}")
 
-
-            all_attr_ops.append(self.translate_parquet_attr(colname, value))
-        return list(itertools.chain(*all_attr_ops, sourceidattr, tsattr))
+            # For cyclical tables, we are either processing only the reference attributes or only the non-reference attributes at a time
+            if self.is_cyclical:
+                if ref_only and colname in self.table.reference_attrs:
+                    all_attr_ops.append(self.translate_parquet_attr(colname, value))
+                elif ref_only or colname in self.table.reference_attrs:
+                    continue
+                else:
+                    all_attr_ops.append(self.translate_parquet_attr(colname, value))
+            else:
+                all_attr_ops.append(self.translate_parquet_attr(colname, value))
```

Comment:
I think you can simplify this conditional:

```suggestion
            if self.is_cyclical and (colname in self.table.reference_attrs) != ref_only:
                continue
            else:
                all_attr_ops.append(self.translate_parquet_attr(colname, value))
```